zxinglight/_zxinglight.cpp

```diff
@@ -65,9 +65,15 @@ static vector<string> *_zxing_read_codes(char *image, int image_size, int width,
 
         Ref<BinaryBitmap> bitmap(new BinaryBitmap(binarizer));
 
-        MultiFormatReader delegate;
-        GenericMultipleBarcodeReader reader(delegate);
-        vector<Ref<Result> > results = reader.decodeMultiple(bitmap, hints);
+        vector<Ref<Result> > results;
+        if(multi) {
+            MultiFormatReader delegate;
+            GenericMultipleBarcodeReader reader(delegate);
+            results = reader.decodeMultiple(bitmap, hints);
+        } else {
+            Ref<Reader> reader(new MultiFormatReader);
```

Comment:
I don't see you cleaning the new `MultiFormatReader` anywhere. Why don't you just allocate it on stack?

Reply:
You're right. I copied that out of zxing-cli and my head wasn't in C++ mode. Thanks for the sharp eyes!!

Reply:
It should be better now!